### PR TITLE
A catalogue row completes itself from what its table printed

### PR DESCRIPTION
**Pull request by jmrplens**
A published porous specimen is almost never complete. A building-materials table gives a bulk density and no porosity. A manufacturer's sheet gives a bulk density and no flow resistivity. Almost nothing gives a characteristic length, because measuring one needs an ultrasonic bench, and `johnson_champoux_allard` wants all five.

This adds the four published relations that close those gaps, each from the page it is printed on.

`porosity_from_bulk_density` is Hopkins Eq. (1.160), PDF page 107 (printed p. 80): the porosity of a fibrous material from its bulk and fibre densities, which holds when the fibres are solid and the binder has negligible mass.

`airflow_resistivity_from_bulk_density` is his Eq. (1.165) on the facing page, the resistivity of a mineral wool from its bulk density and fibre diameter. The constants belong to a material and not to mineral wool in general, so they travel in a `FibreResistivityFit` that carries the fibre diameter they were fitted at and the bulk-density range they were fitted over, and a density outside that range is announced rather than extrapolated in silence. Two are published from Hopkins' page, for the same rock wool measured both ways round, because mineral wool is anisotropic and a flow through the sheet crosses more fibres than one along it.

`viscous_characteristic_length` is Allard & Atalla Eq. (5.25), PDF page 90 (printed p. 80), the viscous length from the three parameters that are measured routinely. It was already in the guide as a closed-form fallback nobody could call.

`fibre_characteristic_lengths` is their Eqs. (5.29) and (5.30), PDF page 91 (printed p. 81), both lengths from a bundle of cylinders. The fibre length per unit volume that those equations are written in is eliminated through its own definition, which is arithmetic on a definition and not a second model.

## What each one is worth

Every one of these is an estimate, so each says how good an estimate it is, measured rather than claimed.

Hopkins' two equations check each other. Put the ends of the bulk-density range the resistivity fit was made over through the porosity relation, at the fibre density he states, and the porosity range he prints beside them comes back: 0,99 at 31 kg/m³ and 0,94 at 155 kg/m³.

Eq. (5.25) is held against every specimen Allard & Atalla print with all four columns at once. There are twenty-three of them, gathered from fourteen pages, and they are now reference data with the table, the PDF page and the printed folio on every row. A unit shape factor lands within a factor of two on eighteen, with the middle of the set near 1,2. The five it misses are the two carpets, the woven screen and two rows whose lengths the book itself took from a different model. So the relation is worth a factor of two on a bulk fibrous or open-cell absorber, and nothing at all on a floor covering.

And the two routes to the viscous length are made to disagree on purpose, on the same rock wool, because they are different models rather than two readings of one: the cylinder bundle is the lower estimate throughout, by 1,4 at 38 kg/m³ and 1,9 at 155 kg/m³. A caller who takes them for two opinions on one quantity will pick the wrong one.

## The rest

The porous guide gains a section in both languages, with the numbers above worked through, and the docs twin gains the same in short. The sentence that used to describe Eq. (5.25) as a formula to type out by hand now names the function.

`check_parameter_units.py` learned `_um`: a micro-geometry is written in micrometres by every book that prints one, and an empirical constant fitted in those units is wrong by twelve orders of magnitude in metres. The tree already published `viscous_length_um` and `thermal_length_um`; the suffix was simply never needed on a watched quantity before.

Sixty-one new tests.

## Summary by CodeRabbit

- **New Features**
  - Added tools for estimating porous-material porosity, airflow resistivity, viscous characteristic length, and fibre characteristic lengths.
  - Added rock-wool reference fits, constants, and directional airflow-resistivity estimates.
  - Added validation and warnings for invalid inputs and densities outside fitted ranges.

- **Documentation**
  - Added API references, equations, examples, validity guidance, and accuracy notes.

- **Tests**
  - Added coverage using published porous-material data, model behavior, boundary conditions, and validation requirements.

### CODE REVIEW

File: src/phonometry/materials/absorbers/porous.py [hunk 1]
```diff
@@ -684,6 +694,241 @@ def limp_frame(
...
+        prints a diameter in micrometres wants half of it, divided by a
+        million.
+    :param bulk_density_kg_m3: Bulk density of the layer, in kg/m3 (> 0).
+    :param fibre_density_kg_m3: Density of the fibre itself, in kg/m3 (> 0).
+    :return: The viscous and thermal lengths, in metres.
+    :raises ValueError: for a non-positive input.
+    """
+    radius = require_positive(fibre_radius_m, "fibre_radius_m")
+    bulk = require_positive(bulk_density_kg_m3, "bulk_density_kg_m3")
+    fibre = require_positive(fibre_density_kg_m3, "fibre_density_kg_m3")
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply src/phonometry/materials/absorbers/porous.py:930]: Applied, and it was a real hole. The ratio of the two densities is the fibre volume fraction, so at or above 1 there is no pore and the cylinder model was returning a positive length for a solid block. It now refuses with a message that says why, which is what `porosity_from_bulk_density` already did, and a test covers both the equality and the greater-than case.

File: site/src/content/docs/materials/absorbers/porous-absorbers.mdx [hunk 2]
```diff
@@ -338,6 +340,93 @@ out far from the flow rig's answer, believe the rig: a fit that has absorbed a
...
+sigma = materials.airflow_resistivity_from_bulk_density(
+    60.0, fit=materials.ROCK_WOOL_LONGITUDINAL_FIT
+)
+phi = materials.porosity_from_bulk_density(
+    60.0, fibre_density_kg_m3=materials.ROCK_WOOL_FIBRE_DENSITY_KG_M3
+)
+print(round(sigma), round(phi, 3))
+# 23226 0.977
+
+# Across the sheet the same wool resists about half as much.
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply site/src/content/docs/materials/absorbers/porous-absorbers.mdx]: Applied, with a third wording rather than the suggested one: "in the plane of the sheet", which is how Hopkins himself puts it on that page and leaves nothing to read two ways. The Spanish comment now says "en el plano de la manta" to match.

File: llms-full.txt [hunk 3]
```diff
@@ -32787,6 +32787,41 @@ worked examples and not a material catalogue: a real specimen is characterised
...
+`FibreResistivityFit` that carries the bulk-density range it was fitted over and
+warning outside it, with the rock wool of his Fig. 1.49 published both lateral
+and longitudinal because mineral wool is anisotropic;
+`viscous_characteristic_length` is Allard & Atalla Eq. (5.25),
+$\Lambda = (8\eta\alpha_\infty/(\sigma\phi))^{1/2}/c$; and
+`fibre_characteristic_lengths` their Eqs. (5.29) and (5.30) for a bundle of
+cylinders, which gives both lengths from the fibre radius and the two densities.
+
+Each says what it is worth. Hopkins' two equations return the porosity range he
+prints beside them, 0,99 at 31 kg/m³ and 0,94 at 155 kg/m³. A unit shape
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply llms-full.txt]: Applied to the docs twin, which is where the error was. The site guide already says it correctly (the bulk-density endpoints go through the porosity relation, not the resistivity one); the twin compressed both equations into one sentence and made it read as though the resistivity fit returned a porosity. The twin now spells out the route.
